File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/LagreFritekstDto.java
package no.nav.melosys.tjenester.gui.dto;

import jakarta.annotation.Nullable;

// Alle felt er nullable: skjemaer som ikke har et fritekstfelt (f.eks. opphørsvedtak uten innledning)
// autolagrer uten det, og autolagring skal aldri feile (MELOSYS-8141).
public record LagreFritekstDto(@Nullable String innledningFritekst,
                               @Nullable String begrunnelseFritekst,
                               @Nullable String trygdeavgiftFritekst) {
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/LagreFritekstDto.kt
package no.nav.melosys.tjenester.gui.dto

@JvmRecord
data class LagreFritekstDto(
    @JvmField val innledningFritekst: String,
    @JvmField val begrunnelseFritekst: String,
    @JvmField val trygdeavgiftFritekst: String
)
